infra/Pos/NtpCheck.hs

```diff
@@ -68,13 +68,13 @@ timeDifferenceWarnInterval = fromIntegral (Infra.ccTimeDifferenceWarnInterval in
 timeDifferenceWarnThreshold :: HasInfraConfiguration => Microsecond
 timeDifferenceWarnThreshold = fromIntegral (Infra.ccTimeDifferenceWarnThreshold infraConfiguration)
 
-type NtpStatusVar = MVar NtpStatus
-
--- Helper to get ntp status
-mkNtpStatusVar :: ( NtpCheckMonad m , Mockables m [ CurrentTime, Delay] )
-    => m NtpStatusVar
-mkNtpStatusVar = do
+-- | Create NTP client, let it work till the first response from servers,
+-- then shutdown and return result.
+getNtpStatusOnce :: ( NtpCheckMonad m , Mockables m [ CurrentTime, Delay] )
+    => m NtpStatus
+getNtpStatusOnce = do
     status <- newEmptyMVar
     let onStatusHandler = putMVar status
-    _ <- ntpSingleShot $ ntpSettings onStatusHandler
```

Comment:
Did it hang? Or just worked for too much time?

Reply:
It worked not enough time 😔 
`ntpSingleShot` stops client after `ntpRequestTimeout`, and NTP client waits for `ntpRequestTimeout` before analyzing responses and executing callback.
Previously client was getting stopped a bit slower than now and it worked, but not now 😁